src/flow_solver/flow_solver_cases/periodic_turbulence.cpp

```diff
@@ -188,18 +189,22 @@ void PeriodicTurbulence<dim, nstate>::output_velocity_field(
     //-------------------------------------------------------------
     std::ofstream FILE (filename);
     
+    const unsigned int higher_poly_degree = this->output_velocity_number_of_subvisions*(dg->max_degree+1)-1; // -1 so that n_quad_pts in 1D is n_subdiv*(P+1)
+    
     // check that the file is open and write DOFs
     if (!FILE.is_open()) {
         this->pcout << "ERROR: Cannot open file " << filename << std::endl;
         std::abort();
     } else if(this->mpi_rank==0) {
-        const unsigned int number_of_degrees_of_freedom_per_state = dg->dof_handler.n_dofs()/nstate;
+        // const unsigned int number_of_degrees_of_freedom_per_state = dg->dof_handler.n_dofs()/nstate;
+        const unsigned int number_of_degrees_of_freedom_per_state = pow(this->number_of_cells_per_direction*(higher_poly_degree+1),dim);
```

Comment:
be careful here, need a virtual getter function for this to accommodate the channel flow grid that has a different number of cells for a given direction